vspace/src/main/java/edu/asu/diging/vspace/core/factory/impl/ImageFactory.java

```diff
@@ -23,4 +35,28 @@ public IVSImage createImage(String filename, String fileType) {
         image.setFileType(fileType);
         return image;
     }
+    
+    /**
+     * Method to rename image   
+     * 
+     * @param imageId - image unique identifier
+     * @param imageForm - ImageForm with updated values for image fields
+     * @return throws FileNotFoundException if no image exists with id, 
+     * throws FileStorageException if file renaming fails 
+     */ 
+    @Override
+    public void editImage(String imageId, ImageForm imageForm) throws FileNotFoundException, FileStorageException {
+        if (imageRepo.findById(imageId).isPresent()) {
+            VSImage image = imageRepo.findById(imageId).get();
```

Comment:
I assume that findById is cached but I'm not sure. Have you checked if this hits the DB once or twice? If it's twice you should move that line up and use the image in your if statement.